plugins/ScheduledReports/Controller.php

```diff
@@ -116,8 +117,8 @@ public function index()
             $savedSegmentsById = array(
                 '' => Piwik::translate('SegmentEditor_DefaultAllVisits')
             );
-            $response = Request::processRequest("SegmentEditor.getAll", ['idSite' => $this->idSite], $defaultRequest = []);
-            foreach ($response as $savedSegment) {
+            $allSegments = SegmentEditor::getAllSegmentsForSite($this->idSite);
```

Comment:
Does this couple the plugins too tightly and could've been the reason it was previously done via an API call?
Does it remove any extensibility here possibly if someone manipulates the API request/response in their plugins?

Reply:
`SegmentEditor::getAllSegmentsForSite` also uses the same API to fetch the data.